src/main/perl/LV.pm

```diff
@@ -98,6 +118,62 @@ sub new_from_system
 
 =pod
 
+=head2 lvcache_ks
+
+Print the ks code for the cache on the logical volume.
+
+=cut
+
+sub lvcache_ks
+{
+    my $self = shift;
+    $self->{cache_lv}->create_ks;
+    my $command = join(" ", LVCONVERT, LVTYPE, 'cache', LVCACHEPOOL,
+       "$self->{volume_group}->{devname}/$self->{cache_lv}->{devname}", "$self->{volume_group}->{devname}/$self->{devname}");
+
+    print <<EOC 
+    if ! lvm lvs $self->{volume_group}->{devname}/$self->{devname} | awk '{ print \$5 }' | grep $self->{cache_lv}->{devname} > /dev/null
+    then
+        $command
+    fi
+EOC
+
+}
+
+
+=pod
+
+=head2 create_cache
+
+Creates the cache on the logical volume. Returns $? (0 if
+success). If the logical volume cache already exists, it returns 0 without
+doing anything.
+
+Returns 0 on success.
+
+=cut
+
+sub create_cache
+{
+    my $self = shift;
+    $self->{cache_lv}->create == 0 or return $?; 
```

Comment:
don't use `$?` here. catch the return code of `create` and return that value explicitly.